consensus/src/quorum_store/utils.rs

```diff
@@ -494,7 +499,14 @@ impl ProofQueue {
                             cur_unique_txns
                                 + txn_summaries
                                     .iter()
-                                    .filter(|txn_summary| !filtered_txns.contains(txn_summary))
+                                    .filter(|txn_summary| {
+                                        !filtered_txns.contains(txn_summary)
+                                                // latest_block_timestamp is microseonds since UNIX epoch
+                                                // expiration_timestamp_secs is seconds since UNIX epoch
+                                                // giving a second buffer for expiration as the expiration time is rounded off to seconds 
+                                            && ((txn_summary.expiration_timestamp_secs + 1) * MICRO_SEC_PER_SEC)
```

Comment:
We should be careful here.
* The multiplication as-is could overflow. So we should instead do integer division.
* We should match the move spec for expiration.

The move spec appears to be
https://github.com/aptos-labs/aptos-core/blob/533f3c33aa0c581da11bf3a14276be8fd40e19c8/aptos-move/framework/aptos-framework/sources/transaction_validation.move#L83-L86

where now_seconds() is the current block timestamp truncated to seconds.

So it appears the correct condition is:
`(self.latest_block_timestamp + 1) / MICRO_SEC_PER_SEC < txn_summary.expiration_timestamp_secs`
because the current block timestamp must be monotonically increasing from the previous timestamp.

If we want a truly accurate expiration condition, we can pass in the current block's timestamp from proposal_generator. I'm not sure that's necessary, as the number of truly unexpired transactions <= unique txn size based on the txn summaries we have.

Reply:
If you are intending to approximate `self.latest_block_timestamp/MICRO_SEC_PER_SEC` with the next integer value, did you mean `(self.latest_block_timestamp + MICRO_SEC_PER_SEC) / MICRO_SEC_PER_SEC < txn_summary.expiration_timestamp_secs`?